lib/internal/modules/cjs/loader.js

```diff
@@ -216,6 +216,11 @@ function tryExtensions(p, exts, isMain) {
   return false;
 }
 
+function readExtensions() {
+  return Object.keys(Module._extensions)
+    .map((ext) => path.extname(ext) || ext);
```

Comment:
Can we avoid the functional array methods and use a regular loop instead?

Reply:
I'm not against this but imo this looks much better than regular loop would. As for the performance implications this is not a hot code and also in the newest v8 AFAIK functional operations have the same or at least similar performance to regular loops.
Though if you want I'll change it. I think it will be as follows.
```js
const exts = Object.keys(Module._extensions);
const result = new Array(exts.length);
for (let i = 0; i < exts.length; ++i)
  result[i] = path.extname(exts[i]) || exts[i];
return result;
```

Reply:
I think you could simplify if further and make it even more efficient than a straight conversion with:

```js
const exts = Object.keys(Module._extensions);
for (let i = 0; i < exts.length; ++i)
  exts[i] = path.extname(exts[i]) || exts[i];
return exts;
```

or avoid reassigning the same value:

```js
const exts = Object.keys(Module._extensions);
for (let i = 0; i < exts.length; ++i) {
  const extName = path.extname(exts[i]);
  if (extName)
    exts[i] = extName;
}
return exts;
```

Reply:
Oh, yeah. Functional thinking got too far =).